src/sentry/scripts/similarity/index.lua

```diff
@@ -0,0 +1,627 @@
+                            local members = redis.call(
+                                'SMEMBERS',
+                                get_bucket_membership_key(
+                                    configuration.scope,
+                                    index,
+                                    time,
+                                    band,
+                                    bucket
+                                )
+                            )
+                            for _, member in ipairs(members) do
+                                candidates[member] = true
+                            end
+                        end
+                    end
+                end
+                return candidates
+            end
+
+            return table.imap(
+                indices,
+                function (index)
+                    local item_frequencies = fetch_bucket_frequencies(index, item_key)
+
+                    local candidates = fetch_candidates(index, item_frequencies)
+                    local candidate_frequencies = {}
+                    for candidate_key, _ in pairs(candidates) do
+                        candidate_frequencies[candidate_key] = fetch_bucket_frequencies(
+                            index,
+                            candidate_key
+                        )
+                    end
+
+                    local results = {}
+                    for key, value in pairs(candidate_frequencies) do
+                        table.insert(
+                            results,
+                            {
+                                key,
+                                table.ireduce(
+                                    table.imap(
+                                        table.izip(
+                                            item_frequencies,
+                                            value
+                                        ),
+                                        function (v)
+                                            local dist = get_manhattan_distance(
+                                                scale_to_total(v[1]),
+                                                scale_to_total(v[2])
+                                            )
+                                            return 1 - (dist / 2)
+                                        end
+                                    ),
+                                    function (total, item)
+                                        return total + item
+                                    end,
+                                    0
+                                ) / configuration.bands
+                            }
+                        )
+                    end
+
+                    table.sort(
+                        results,
+                        function (left, right)
+                            return left[2] > right[2]
+                        end
+                    )
```

Comment:
Sorting this here feels like the right responsibility for the database, but I might push it back up to the application for performance reasons.